src/layman/upgrade/upgrade_v1_23.py

```diff
@@ -168,3 +168,26 @@ def delete_user_roles():
     role_not_exists = response.status_code == 404
     if not role_not_exists:
         response.raise_for_status()
+
+
+def restrict_workspace_name_length():
+    logger.info(f'    Restrict username length')
+
+    select_too_long = f"""
+select name
+from {settings.LAYMAN_PRIME_SCHEMA}.workspaces
+where length(name) > 59
+;"""
+    too_long_workspace_name = db_util.run_query(select_too_long)
+    if len(too_long_workspace_name) > 0:
+        raise NotImplementedError(f"Too long workspace names: {[name[0] for name in too_long_workspace_name]}")
+
+    # For direct ALTER TABLE raises "ERROR: cannot alter type of a column used by a view or rule"
```

Comment:
It can be probably used if we switch migrations create_role_service_schema and restrict_workspace_name_length